task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -87,12 +96,13 @@ def retry_loop(self):
 
             time.sleep(1)
 
-    def run(self, task_config):
-        if task_config.task_id not in self.task_retries:
+    def run(self, task_config, task_id):
+        if task_id not in self.task_retries:
             with self.task_retries_lock:
                 self.task_retries = self.task_retries.set(
-                    task_config.task_id, self.retries)
-        self.executor.run(task_config)
+                    task_id, self.retries)
+
+        self.executor.run(task_config, self.grow(task_id))
```

Comment:
default interface of the runner has optional task_id, so if user passes the id that has already been used - they supposedly know what they are doing. the effect is the same as if user had called .run with the identical task_config using the current master branch.

but it's a good point, this is undocumented and undefined behavior. we should decide what do we really want to happen when user does this. do we want to force another retry if the task is still running? do we want to restart from 0? maybe only if task had already finished?